### PR TITLE
1221 Search for case and show results

### PR DESCRIPTION
**Pull request by AdamHawtin**
# Motivation and Context

The first feature needed for ops-ui is to be able to search for cases via a postcode. This PR adds that functionality as well as IAP Audit logging and adding the CDN to ops-ui.

# What has changed

- Search for cases via postcode
- IAP audit logging
- Added flask/jinja templates to show postcode results.
- Added Flask blueprints for routing
- Now runs with Gunicorn

# How to test?

- Run with the other branches on the ticket. When you have cases in your database, you should be able to run the ops-ui and find cases with a postcode.
- Try running this in your GCP environment and see if you can connect to it.
- Anyone testing I can add to the IAP so you could try and connect to the domain set up in `census-rm-ryangrundy`

# Links

[Trello](https://trello.com/c/qkkjvupd)

### CODE REVIEW

**Review by NickGrantX — CHANGES_REQUESTED** [review]
Missing columns, specified in the acceptance criteria (although I personally think it's far too many):
 - org name
 - case type
 - estab type
 - UPRN
 - region

File: census_rm_ops_ui/templates/layouts/base-head.html [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+{% set cdn_hash = "v1.8.1" %}
+{% set cdn_url_prefix = "https://cdn.ons.gov.uk/sdc/"~cdn_hash %}
+{% set css_responsive_cdn = cdn_url_prefix~"/css/responsive.css" %}
+{% set scripts_cdn = cdn_url_prefix~"/scripts/bundle.js" %}
+
+<html lang="en" class="no-js">
+  <head>
+    <meta charset="utf-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge"/>
+    <meta name="description" content="ONS Social Surveys">
```
Comment (NickGrantX) [inline census_rm_ops_ui/templates/layouts/base-head.html]: needs updating

File: census_rm_ops_ui/templates/layouts/base-head.html [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    <meta name="author" content="Survey Data Collection team for ONS" />
```
Comment (NickGrantX) [inline census_rm_ops_ui/templates/layouts/base-head.html]: needs updating

File: census_rm_ops_ui/templates/layouts/base-head.html (hunk 1 above)
Comment (ryangrundy7) [reply census_rm_ops_ui/templates/layouts/base-head.html]: Good spot 👍

File: census_rm_ops_ui/templates/layouts/base-head.html (hunk 2 above)
Comment (ryangrundy7) [reply census_rm_ops_ui/templates/layouts/base-head.html]: Good spot 👍

File: census_rm_ops_ui/templates/layouts/base.html [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        </div>
+
+        <header class="header header--internal header--thin">
+          {% include 'partials/header.html' %}
+          {% block header %}{% endblock header %}
+        </header>
+
+        <div class="container container--wide page__container">
+        {% if g.maintenance_message %}
+{#          {% include 'partials/planned-maintenance-message.html' %}#}
```
Comment (NickGrantX) [inline census_rm_ops_ui/templates/layouts/base.html]: commented out?

File: census_rm_ops_ui/views/postcode_search.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
+from flask import Blueprint, current_app, render_template, request
+
+from census_rm_ops_ui.controllers import case_controller
+
+ADDRESS_SUMMARY_FIELDS = ('organisationName',
```
Comment (NickGrantX) [inline census_rm_ops_ui/views/postcode_search.py:5]: I don't think this is meeting the acceptance criteria, although there are far too many columns IMHO

**Comment by ryangrundy7** [comment]
> Missing columns, specified in the acceptance criteria (although I personally think it's far too many):
> 
> * org name
> * case type
> * estab type
> * UPRN
> * region

@NickGrantONS Org name is part of the address summary that gets put into the table. Estab type and case type are included in the page. UPRN and Region aren't in the prototype that we were basing this on. Could be a question for Dan if they need to be included.

Spoke to Dan, business want those two extra so I'll add them 👍

File: census_rm_ops_ui/templates/layouts/base.html (hunk 3 above)
Comment (ryangrundy7) [reply census_rm_ops_ui/templates/layouts/base.html]: That section can be removed!

File: census_rm_ops_ui/views/postcode_search.py (hunk 4 above)
Comment (ryangrundy7) [reply census_rm_ops_ui/views/postcode_search.py:5]: Should be matching the AC now

**Review by NickGrantX — APPROVED** [review]

**Review by liamtoozer — APPROVED** [review]
Nice work! 👍

**Pull request merged**